User flow language customization documentation (#10918)
microsoftgraph/microsoft-graph-docs: api-reference/beta/api/b2cidentityuserflow-list-languages.md

---
title: "List languages"
description: "Retrieve a list of languages supported for customization within a B2C user flow."
author: "jkdouglas"
localization_priority: Normal
ms.prod: "microsoft-identity-platform"
doc_type: apiPageType
---

# List languages

Namespace: microsoft.graph

> Retrieve a list of languages supported for customization in a B2C user flow.

Reviewer: I think it should probably be Azure AD B2C.

> **Note:** To retrieve a list of languages supported for customization, you must first enable language customization on your B2C user flow. See the [Update b2cIdentityUserFlow](../api/b2cidentityuserflow-update.md) topic for an example.

Reviewer: ```suggestion
**Note:** To retrieve a list of languages supported for customization, you must first enable language customization on your Azure AD B2C user flow. For more information, see [Update b2cIdentityUserFlow](../api/b2cidentityuserflow-update.md).
```

## Permissions

One of the following permissions is required to call this API. To learn more, including how to choose permissions, see [Permissions](/graph/permissions-reference).

|Permission type      | Permissions (from least to most privileged)              |
|:--------------------|:---------------------------------------------------------|
|Delegated (work or school account)|IdentityUserFlow.Read.All, IdentityUserFlow.ReadWrite.All|
|Delegated (personal Microsoft account)| Not supported.|
|Application|IdentityUserFlow.Read.All, IdentityUserFlow.ReadWrite.All|

The work or school account needs to belong to one of the following roles:

* Global administrator
* External Identity User Flow administrator

## HTTP request

<!-- {
  "blockType": "ignored"
}
-->

``` http
GET /identity/b2cUserFlows/{id}/languages
```

## Optional query parameters

> This method supports the `$filter` for filtering the results to show only the enabled languages. For general information, see [OData query parameters](/graph/query-parameters).

Reviewer: ```suggestion
This method supports the `$filter` query parameter to show only the enabled languages. For general information, see [OData query parameters](/graph/query-parameters).
```

## Request headers

|Name|Description|
|:---|:---|
|Authorization|Bearer {token}. Required.|

## Request body

Do not supply a request body for this method.

## Response

If successful, this method returns a `200 OK` response code and a collection of [userFlowLanguageConfiguration](../resources/userflowlanguageconfiguration.md) objects in the response body.

## Examples

### Example 1: Retrieve a list of all languages

> #### Request

Reviewer: Add the sentence: The following is an example of the request.

<!-- {
  "blockType": "request",
  "name": "get_userflowlanguageconfiguration"
}
-->

``` http
GET https://graph.microsoft.com/beta/identity/b2cUserFlows/B2C_1_CustomerSignUp/languages
```

> #### Response

Reviewer: Add the sentence: The following is an example of the response.

**Note:** The response object shown here might be shortened for readability.
<!-- {
  "blockType": "response",
  "truncated": true,
  "@odata.type": "Collection(microsoft.graph.userFlowLanguageConfiguration)"
}
-->

``` http
HTTP/1.1 200 OK
Content-Type: application/json

{
  "odata.context": "https://graph.microsoft.com/beta/$metadata#identity/b2cUserFlows('B2C_1_CustomerSignUp')/languages",
  "value": [
    {
      "id": "en",
      "isEnabled": true,
      "displayName": "English"
    },
    {
      "id": "de",
      "isEnabled": false,
      "displayName": "Deutsch"
    }
  ]
}
```

### Example 2: Retrieve a list of only enabled languages

> #### Request

Reviewer: Add the sentence: The following is an example of the request.

<!-- {
  "blockType": "request",
  "name": "get_userflowlanguageconfiguration_filter"
}
-->

``` http
GET https://graph.microsoft.com/beta/identity/b2cUserFlows/B2C_1_CustomerSignUp/languages?$filter=isEnabled eq true
```

> #### Response

Reviewer: Add the sentence: The following is an example of the response.
